File: rb/features.rb
# WaterQualityArchive SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module WaterQualityArchiveFeatures
  def self.make_feature(name)
    case name
    when "base"
      WaterQualityArchiveBaseFeature.new
    when "test"
      WaterQualityArchiveTestFeature.new
    else
      WaterQualityArchiveBaseFeature.new
    end
  end
end

File: py/features.py
# WaterQualityArchive SDK feature factory

from feature.base_feature import WaterQualityArchiveBaseFeature
from feature.test_feature import WaterQualityArchiveTestFeature


def _make_feature(name):
    features = {
        "base": lambda: WaterQualityArchiveBaseFeature(),
        "test": lambda: WaterQualityArchiveTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
